l1-contracts/contracts/l2-system/zksync-os/ZKOSContractHelper.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: MIT
+// We use a floating point pragma here so it can be used within other projects that interact with the ZKsync ecosystem without using our exact pragma version.
+pragma solidity ^0.8.20;
+
+/**
+ * @author Matter Labs
+ * @custom:security-contact security@matterlabs.dev
+ * @notice Smart contract for sending arbitrary length messages to L1
+ * @dev by default ZkSync can send fixed-length messages on L1.
+ * A fixed length message has 4 parameters `senderAddress`, `isService`, `key`, `value`,
+ * the first one is taken from the context, the other three are chosen by the sender.
+ * @dev To send a variable-length message we use this trick:
+ * - This system contract accepts an arbitrary length message and sends a fixed length message with
+ * parameters `senderAddress == this`, `isService == true`, `key == msg.sender`, `value == keccak256(message)`.
+ * - The contract on L1 accepts all sent messages and if the message came from this system contract
+ * it requires that the preimage of `value` be provided.
+ */
+interface IL1Messenger {
```

Comment:
this is a lot of duplication. We already have most of these implemented. If needed please create separate IL2ToL1MessengerEra and IL2ToL1MessengerZKSyncOs. 

Also please one interface per file